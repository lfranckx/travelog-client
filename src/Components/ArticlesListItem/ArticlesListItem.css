.main-page-articles article {
    padding: 10px 0;
    border-bottom: 1px solid lightgrey;
}

.article-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-bottom: 5px;
}

.article-container > * {
    margin: 5px;
}

.author {
    color: #03a87c;
    font-size: 18px;
}

.author-profile {
    height: 50px;
    width: 50px;
    border-radius: 50%;
}

.thumbnail {
    width: 25%;
    height: 25%;
    min-height: 100px;
    max-height: 420px;
    min-width: 100px;
    max-width: 500px;
    background-size: cover;
    background-position: 50% 50%;
}