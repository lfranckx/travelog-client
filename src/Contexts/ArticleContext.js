/*eslint semi: ["error", "always"]*/
import React, { Component } from 'react';

export const nullArticle = {
    author: {},
    tags: [],
};
export const nullAuthor = {};
export const nullUser = {};
export const nullUsersArticles = [];

const ArticleContext = React.createContext({
    error: null,
    article: nullArticle,
    author: nullAuthor,
    user: nullUser,
    articlesList: [],
    authorsList: [],
    usersArticles: nullUsersArticles,
    setError: () => {},
    clearError: () => {},
    setArticle: () => {},
    clearArticle: () => {},
    setAuthor: () => {},
    clearAuthor: () => {},
    setUser: () => {},
    clearUser: () => {},
    setArticlesList: () => {},
    setAuthorsList: () => {},
    setUsersArticles: () => {},
    clearUsersArticles: () => {}
});
export default ArticleContext;

export class ArticleProvider extends Component {
    state = {
        error: null,
        article: nullArticle,
        author: nullAuthor,
        user: nullUser,
        articlesList: [],
        authorsList: [],
        usersArticles: []
    };

    setError = error => {
        console.error(error);
        this.setState({ error });
    };

    clearError = () => {
        this.setState({ error: null });
    };

    setArticle = article => {
        this.setState({ article });
    };

    clearArticle = () => {
        this.setArticle(nullArticle);
    };

    setAuthor = author => {
        this.setState({ author });
    };

    clearAuthor = () => {
        this.setAuthor(nullAuthor);
    };

    setUser = user => {
        console.log('setting user:', user);
        this.setState({ user });
    };

    clearUser = () => {
        this.setState(nullUser);
    };

    setArticlesList = articlesList => {
        this.setState({ articlesList });
    };

    setAuthorsList = authorsList => {
        this.setState({ authorsList });
    };

    setUsersArticles = usersArticles => {
        this.setState({ usersArticles });
    };

    clearUsersArticles = () => {
        this.setUsersArticles(nullUsersArticles);
    };

    render() {
        const value = {
            error: this.state.error,
            article: this.state.article,
            author: this.state.author,
            user: this.state.user,
            articlesList: this.state.articlesList,
            authorsList: this.state.authorsList,
            usersArticles: this.state.usersArticles,
            setError: this.setError,
            clearError: this.clearError,
            setArticle: this.setArticle,
            clearArticle: this.clearArticle,
            setAuthor: this.setAuthor,
            clearAuthor: this.clearAuthor,
            setUser: this.setUser,
            clearUser: this.clearUser,
            setArticlesList: this.setArticlesList,
            setAuthorsList: this.setAuthorsList,
            setUsersArticles: this.setUsersArticles,
            clearUsersArticles: this.clearUsersArticles
        };
        
        return (
            <ArticleContext.Provider value={value}>
                {this.props.children}
            </ArticleContext.Provider>
        );
    }   
}