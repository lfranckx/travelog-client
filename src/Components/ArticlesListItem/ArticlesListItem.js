import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './ArticlesListItem.css';

export default class ArticlesListItem extends Component {
    
    render() {
        const { article, authors } = this.props;

        let authorObject;
        authors.map(author => {
            if (author.author_id === article.user_id)
                authorObject = author; 
            return authorObject;
        });

        if (!authorObject) {
            return <div className="loading">Loading...</div>
        }
        
        return (
                <article>
                    <div>{article.date}</div>
                    <Link to={`/author/${authorObject.author_id}`}>
                        <div>{article.author}</div>
                    </Link>
                    <Link to={`/article/${article.id}`}>
                        <img className="thumbnail" src={article.image_url} alt="thumbnail" />
                        <h3>{article.title}</h3>
                        <h4>{article.description}</h4>
                        <p>Read more...</p>
                    </Link>
                </article>
            
        );
    }
}