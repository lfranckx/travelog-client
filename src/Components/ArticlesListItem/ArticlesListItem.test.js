import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ArticlesListItem from './ArticlesListItem';
import { ArticleProvider } from '../../Contexts/ArticleContext';

describe(`ArticlesListItem`, () => {
    const props = {
        articles: {
            author: 'Sam Smith',
            body: 'And they say in truth that a man is made of desire. As his desire is, so is his faith. As his faith is, so are his works. As his works are, so he becomes. â€” The Supreme Teaching of the Upanishads',
            date: "Mon May 18 2020",
            description: "Favorite quotes",
            id: 1,
            image_url: 'http://placehold.it/500x500',
            profile_image: 'http://placehold.it/500x500',
            title: 'Vagabonding',
            user_id: 1,
            username: 'user1'
        },
        authors: [
            {
                about: 'This is my about me.',
                name: 'Sam Smith',
                profile_image: 'http://placehold.it/500x500',
            }
        ]
    };
    

    it(`renders an article given props`, () => {
        const wrapper = shallow(
            <ArticleProvider>
                <ArticlesListItem {...props}/>
            </ArticleProvider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});