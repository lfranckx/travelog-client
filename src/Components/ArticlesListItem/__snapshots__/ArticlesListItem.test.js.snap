// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesListItem renders an article given props 1`] = `
<ContextProvider
  value={
    Object {
      "active": false,
      "addComment": [Function],
      "article": Object {},
      "articlesList": Array [],
      "author": Object {},
      "authorsList": Array [],
      "clearArticle": [Function],
      "clearAuthor": [Function],
      "clearError": [Function],
      "clearUser": [Function],
      "clearUsersArticles": [Function],
      "comments": Array [],
      "error": null,
      "search": "",
      "setArticle": [Function],
      "setArticlesList": [Function],
      "setAuthor": [Function],
      "setAuthorsList": [Function],
      "setComments": [Function],
      "setError": [Function],
      "setUser": [Function],
      "setUsersArticles": [Function],
      "toggleActive": [Function],
      "updateSearch": [Function],
      "user": Object {},
      "usersArticles": Array [],
    }
  }
>
  <ArticlesListItem
    articles={
      Object {
        "author": "Sam Smith",
        "body": "And they say in truth that a man is made of desire. As his desire is, so is his faith. As his faith is, so are his works. As his works are, so he becomes. â€” The Supreme Teaching of the Upanishads",
        "date": "Mon May 18 2020",
        "description": "Favorite quotes",
        "id": 1,
        "image_url": "http://placehold.it/500x500",
        "profile_image": "http://placehold.it/500x500",
        "title": "Vagabonding",
        "user_id": 1,
        "username": "user1",
      }
    }
    authors={
      Array [
        Object {
          "about": "This is my about me.",
          "name": "Sam Smith",
          "profile_image": "http://placehold.it/500x500",
        },
      ]
    }
  />
</ContextProvider>
`;
